### PR TITLE
[JENKINS-72016] Define a thread pool for `ProxyConfiguration`’s `HttpClient`

### PR DESCRIPTION
**Pull request by jglick**
#7398 advertises a standardized `HttpClient` factory, and says

> The returned instances of `HttpClient` […] are intended to be short-lived, not to be stored in memory in some sort of cache […].

When this advice is followed in some CloudBees CI plugins, obtaining a fresh `HttpClient` per request, under some circumstances where many requests are made the result is a gigantic number of threads with `HttpClient` in the name, eventually crashing the controller.

See [JENKINS-72016](https://issues.jenkins.io/browse/JENKINS-72016).

### Testing done

Ran the new test with

```bash
watch 'jstack $(jps -lm | fgrep booter | awk "{print \$1}")|egrep \^\"|wc -l'
```

Prior to patch: the test slows down and finally hangs without making it to 9000 requests. There are in the vicinity of 18000 threads in the JVM at this point. Sometimes the JVM crashes with `OutOfMemoryError` claiming it cannot create a new native thread. (This is pre-Loom!)

After patch: the test runs steadily to completion. The JVM thread count fluctuates around 4000 or so. A new `HttpClientImpl.SelectorManager` thread is created each time, but these seem to terminate themselves fairly soon rather than piling up indefinitely.

### Proposed changelog entries

- Do not create a large number of threads when making numerous HTTP requests.

### Proposed upgrade guideliness

N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8490"></a>

### CODE REVIEW

File: test/src/test/java/hudson/ProxyConfigurationTest.java
Comment (jglick) [inline test/src/test/java/hudson/ProxyConfigurationTest.java:47]: Takes <1m on my laptop, so I am hoping this is reasonable to keep as is. If not, it could be `@Ignore`d or deleted.

**Review by Vlatombe — APPROVED** [review]
Looks reasonable and fixes the performance issue when using a lot of throw-away `HttpClients`

**Review by Vlatombe — APPROVED** [review]
Looks reasonable and fixes the performance issue when using a lot of throw-away `HttpClients`

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by timja** [comment]
(removed ready for merge, the new test is failing on windows)

**Review by Dohbedoh — APPROVED** [review]

**Review by jtnord — APPROVED** [review]

**Pull request merged**
